### ISSUE TITLE
Health enum default of 'Green' is counter-intuitive for error scenarios

### ISSUE THREAD

**Issue by lakario** [elastic/elasticsearch-net#3452]
**NEST/Elasticsearch.Net version**:
6.3.1

**Elasticsearch version**:
Any

**Description of the problem including expected versus actual behavior**:

When invoking the `ClusterHealth()` and `ClusterHealthAsync()` methods with an invalid or unconnectable Elasticsearch endpoint, the `IClusterHealthResponse.Status` enum property remains assigned the value of `Green`. Given that the call failed, the Status was never retrieved and thus returning `Health.Green`, here is at best counter-intuitive.  This `Status` property should either be nullable, or the enum should have another zero state to represent that the value was not retrieved. 

Ideally, `Green` should not be the default value for the `Health` enum.

**Steps to reproduce**:
 1. Configure an `ElasticClient` with an invalid or unreachable cluster
 2. Invoke `IElasticClient.ClusterHealthyAsync()`
 3. Examine `IClusterHealthResponse.Status`

**Describe the feature**:

**Comment by russcam** [comment]
For an invalid or unconnectable cluster, the `.IsValid` property on the response will be `false`, or if the client is configured to throw exceptions, an exception will be thrown if there is an issue. In both scenarios, the value of `Status` should not be used on the Cluster Health API.

> This Status property should either be nullable, or the enum should have another zero state to represent that the value was not retrieved.

The response maps those values that can be returned from Elasticsearch, so I think adding another client specific value would be confusing. Nullable would work, but given the validity semantics, I'm not sure that it provides additional value.

**Comment by lakario** [comment]
I fully agree that `.IsValid` is the correct path to a successful implementation

**Comment by lakario** [comment]
I agree, nullable would be a better option than to introduce a new enum value, but unfortunately that is liable to introduce a breaking API change. Prior to switching to an enum, presumably the `string` for `Status` was left null?

**Comment by russcam** [comment]
There would always be a value for status in the response as far as I can see, as it maps to the `ClusterHealthStatus` enum in Elasticsearch:

https://github.com/elastic/elasticsearch/blob/5be79ed02ceb9fbe6c867a888ada2af8d53fd8d2/core/src/main/java/org/elasticsearch/cluster/health/ClusterHealthStatus.java#L32-L35

and

https://github.com/elastic/elasticsearch/blob/282ab0d89ec5789d1020247b6b79b5578c843c0f/core/src/main/java/org/elasticsearch/action/admin/cluster/health/ClusterHealthResponse.java#L236

**Comment by lakario** [comment]
The scenario I reported above however is one where there is no response due to a failed connection. As you've stated, `.IsValid` would be the correct property for an implementation to refer to. No doubt, there are many similar situations throughout the NEST framework which we be subject to the same problem in the event of an invalid connection. 

If you like feel free to file this issue under _enhancement_.
